azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -781,3 +790,42 @@ def _remove_secrets_from_secret_sync(
                     resource=secret_sync,
                 )
                 return wait_for_terminal_state(poller)
+
+    def _get_secret_names(self, keyvault_name: str) -> List[str]:
+        secret_iteratable = self.keyvault_client.get_secrets(
+            vault_base_url=KEYVAULT_URL.format(keyvaultName=keyvault_name)
+        )
+        return [secret.get("id") for secret in secret_iteratable]
+
+    def _begin_delete_secret(self, keyvault_name: str, secret_name: str):
+        # Construct vault URL
+        vault_url = KEYVAULT_URL.format(keyvaultName=keyvault_name)
+
+        # Initiate deletion
+        pipeline_response = self.keyvault_client.delete_secret(
+            vault_base_url=vault_url,
+            secret_name=secret_name,
+            cls=lambda pipeline_response, _, __: pipeline_response,
+        )
+
+        for attempt in range(SECRET_DELETE_MAX_RETRIES):
+            try:
+                # Check if secret is deleted
+                self.keyvault_client.get_deleted_secret(
+                    vault_base_url=vault_url,
+                    secret_name=secret_name,
+                )
+                return  # Exit if no exception, deletion confirmed
+            except ResourceNotFoundError:
+                # Secret not yet deleted; retry after delay
+                transport: HttpTransport = cast(HttpTransport, pipeline_response.context.transport)
+                transport.sleep(SECRET_DELETE_RETRY_INTERVAL)
+                attempt += 1
+            except HttpResponseError as e:
+                if e.status_code == 403:
+                    # Permission issue encountered; exit loop
+                    break
+                raise
+
+        # Failed to confirm deletion after retries
+        raise TimeoutError(f"Failed to delete secret '{secret_name}' within {SECRET_DELETE_MAX_RETRIES} retries.")
```

Comment:
Good thought on TimeoutError, while Python exceptions are nice is there a suitable exception to raise in azure.cli.core.azclierror? This has some advantage in telemetry and a stack trace isn't shown.

Reply:
got it! I found one under azclierror that seems to be suitable for timeout issue
<img width="250" alt="image" src="https://github.com/user-attachments/assets/a5f00870-d3cb-41bd-94db-e07bfe0ddd44">

Reply:
AzureConnectionError is not my favorite as that description indicates a client connection issue, rather than timing out waiting for an operation to complete when the connection itself is not at fault.

Reply:
Lets revert to raise TimeoutError for this PR